File: app/src/main/java/tech/thdev/app/view/weekboxoffice/WeeklyBoxOfficeFragment.java
package tech.thdev.app.view.weekboxoffice;

import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.RelativeLayout;
import android.widget.TextView;

import org.jetbrains.annotations.Nullable;

import java.text.SimpleDateFormat;

import butterknife.BindView;
import butterknife.OnClick;
import tech.thdev.app.R;
import tech.thdev.app.adapter.movie.adapter.BoxOfficeAdapter;
import tech.thdev.app.data.source.MovieSearchRepository;
import tech.thdev.app.view.weekboxoffice.presenter.WeeklyBoxOfficeContract;
import tech.thdev.app.view.weekboxoffice.presenter.WeeklyBoxOfficePresenter;
import tech.thdev.base.view.BasePresenterFragment;

/**
 * Created by tae-hwan on 12/13/16.
 */

public class WeeklyBoxOfficeFragment extends BasePresenterFragment<WeeklyBoxOfficeContract.View, WeeklyBoxOfficeContract.Presenter> implements WeeklyBoxOfficeContract.View {

    @BindView(R.id.recycler_view)
    RecyclerView recyclerView;

    @BindView(R.id.rl_empty_view)
    RelativeLayout rlEmptyView;

    @BindView(R.id.tv_no_content)
    TextView tvNoContent;

    @Nullable
    @Override
    public WeeklyBoxOfficeContract.Presenter onCreatePresenter() {
        return new WeeklyBoxOfficePresenter();
    }

    @Override
    public int getLayout() {
        return R.layout.fragment_box_office;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        BoxOfficeAdapter boxOfficeAdapter = new BoxOfficeAdapter(getContext());
        recyclerView.setAdapter(boxOfficeAdapter);

        getPresenter().setSampleOneModel(boxOfficeAdapter);
        getPresenter().setSampleOneView(boxOfficeAdapter);

        getPresenter().setMovieSearchRepository(MovieSearchRepository.INSTANCE);

        getPresenter().getWeeklyBoxOffice(getContext(), new SimpleDateFormat("yyyyMMdd").format(System.currentTimeMillis() - ((1000 * 60 * 60 * 24) * 7)));
    }

    @Override
    public boolean isFinishing() {
        return getActivity().isFinishing();
    }

    @Override
    public void loadFail(String message) {
        rlEmptyView.setVisibility(View.VISIBLE);
        tvNoContent.setText(message);
    }

    @OnClick(R.id.btn_reload)
    public void onClickReload(View view) {
        getPresenter().getWeeklyBoxOffice(getContext(), new SimpleDateFormat("yyyyMMdd").format(System.currentTimeMillis() - ((1000 * 60 * 60 * 24) * 7)));
    }
}

File: app/src/main/java/tech/thdev/app/ui/weekboxoffice/WeeklyBoxOfficeFragment.kt
package tech.thdev.app.ui.weekboxoffice

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import tech.thdev.app.R
import tech.thdev.app.base.BasePresenterFragment
import tech.thdev.app.data.source.MovieSearchRepository
import tech.thdev.app.common.adapter.BoxOfficeAdapter
import tech.thdev.app.ui.weekboxoffice.presenter.WeeklyBoxOfficeContract
import tech.thdev.app.ui.weekboxoffice.presenter.WeeklyBoxOfficePresenter
import java.text.SimpleDateFormat

/**
 * Created by tae-hwan on 12/13/16.
 */
class WeeklyBoxOfficeFragment :
    BasePresenterFragment<WeeklyBoxOfficeContract.View, WeeklyBoxOfficeContract.Presenter>(),
    WeeklyBoxOfficeContract.View {

    companion object {
        fun newInstance(): WeeklyBoxOfficeFragment =
            WeeklyBoxOfficeFragment()
    }

    private val boxOfficeAdapter by lazy {
        BoxOfficeAdapter()
    }

    override fun onCreatePresenter(): WeeklyBoxOfficeContract.Presenter =
        WeeklyBoxOfficePresenter(
            boxOfficeAdapter,
            boxOfficeAdapter,
            MovieSearchRepository
        )

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_box_office, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view.findViewById<RecyclerView>(R.id.recycler_view).run {
            adapter = boxOfficeAdapter
        }

        view.findViewById<Button>(R.id.btn_reload).setOnClickListener {
            loadItem()
        }

        loadItem()
    }

    private fun loadItem() {
        presenter.getWeeklyBoxOffice(
            SimpleDateFormat("yyyyMMdd")
                .format(System.currentTimeMillis() - 1000 * 60 * 60 * 24 * 7)
        )
    }

    override fun loadFail(messageRes: Int) {
        view?.findViewById<TextView>(R.id.tv_no_content)?.setText(messageRes)
        view?.findViewById<View>(R.id.rl_empty_view)?.visibility = View.VISIBLE
    }
}